snapcraft/providers.py

```diff
@@ -42,6 +42,7 @@
     "core18": bases.BuilddBaseAlias.BIONIC,
     "core20": bases.BuilddBaseAlias.FOCAL,
     "core22": bases.BuilddBaseAlias.JAMMY,
+    "core24": bases.BuilddBaseAlias.DEVEL,
     "devel": bases.BuilddBaseAlias.DEVEL,
```

Comment:
@mr-cal we might need to clean this up, devel is only a build-base but never a base

Reply:
https://github.com/snapcore/snapcraft/issues/4539